email_from_template/send_mail.py

```diff
@@ -74,6 +74,29 @@
     </a>
     {% endblock %}
 
+:mod:`django-email-from-template` also has the ability to only send the email
+when the current transaction commits. Pass ``on_commit=True`` to ``send_mail``
+to enable this functionality - for backwards compatability and predictability
+it is not enabled by default. Note that the rendering still occurs during the
+transaction, it is only the sending that is deferred - this is desirable as it
+means that failing to render a template will still result in a transaction
+rolling back.::
+
+    from email_from_template import send_mail
+
+    send_mail([user.email], 'path/to/my_email.email', {
+        'user': user,
+        'num_posts': num_posts,
+    }, on_commit=True)
+
+The return value of ``send_mail`` is, by default, the number of messages sent.
+When passed ``on_commit=True``, we cannot know the number of messages that send
+successfully until after the transaction commits, so we return ``None``. We
+don't return 0 as this could indicate an explicit error (which has not
+occurred), however care should be taken when checking this return code, to
+ensure that 0 is checked for explicitly, as ``None`` is falsy.
```

Comment:
Yeah, I think given that this behaviour is new, and not the default, returning `None` here is an ok thing to do, and in my opinion, the right choice.

However, another option would be to wrap `send_mail` with a new function that handles the `on_commit` functionality, which can therefore have a different return type. Would you prefer that?